docs/design/adrs/2024-04-08-use-pull-throughout-for-ingestion.md

```diff
@@ -0,0 +1,47 @@
+# Use Pull approach(Integrated Vectorization) throughout for ingestion
```

Comment:
Can we also create a short doc (not as part of this PR but as part of the IV work in general) that lists the pros and cons of Push vs Pull as a way to help our users make the right choice of push vs pull for their specific use case. 

e.g. Cost considerations, ease of maintainability, etc. What factors do the users need to consider when making a choice.

Reply:
Sure